### PR TITLE
Eapi updates

### PR DESCRIPTION
**Pull request by SoapGentoo**

### CODE REVIEW

File: app-text/docbook-xsl-stylesheets/docbook-xsl-stylesheets-1.79.1-r3.ebuild [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Copyright 1999-2022 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=6
-USE_RUBY="ruby25 ruby26 ruby27"
+EAPI=8
 
+USE_RUBY="ruby25 ruby26 ruby27"
```
Comment (thesamesam) [inline app-text/docbook-xsl-stylesheets/docbook-xsl-stylesheets-1.79.1-r3.ebuild:6]: drop 25

File: media-gfx/xsane/xsane-0.999-r4.ebuild [hunk 2]
```diff
@@ -41,24 +39,24 @@ src_prepare() {
 
 	# Apply multiple fixes from different distributions
 	# Drop included patch and reuse patchset from prior version
-	rm "${WORKDIR}/${PN}-0.998-patches-2"/005-update-param-crash.patch || die
-	epatch "${WORKDIR}/${PN}-0.998-patches-2"/*.patch
+	rm "${WORKDIR}/${PN}-0.998-patches-3"/005-update-param-crash.patch || die
```
Comment (thesamesam) [inline media-gfx/xsane/xsane-0.999-r4.ebuild:42]: Didn't you repack it? 🤣

File: media-gfx/xsane/xsane-0.999-r4.ebuild [hunk 3]
```diff
@@ -77,14 +75,14 @@ src_install() {
 	# link xsane so it is seen as a plugin in gimp
 	if use gimp; then
 		local plugindir
-		local gimptool=( "${EPREFIX}"/usr/bin/gimptool* )
+		local gimptool=( "${ESYSROOT}"/usr/bin/gimptool* )
```
Comment (thesamesam) [inline media-gfx/xsane/xsane-0.999-r4.ebuild:78]: I think EPREFIX might be right for plugins?

Comment (SoapGentoo) [reply media-gfx/xsane/xsane-0.999-r4.ebuild:75]: yes, and if it's from a prefix installation, it will still work?

**Comment by dm0-** [comment]
@mattst88 The issue is https://bugs.gentoo.org/723112 specifically, while EAPI 6 seemed to not cause breakage with `MULTILIB_USEDEP` in `DEPEND`.  Don't let me impede progress if I'm the only one affected, though, I will work around it locally.

**Comment by SoapGentoo** [comment]
@dm0- what if we drop rust into `DEPEND` again. Would that help?

**Comment by dm0-** [comment]
@SoapGentoo If Rust is in `DEPEND` with EAPI 7 or 8, it would get cross-compiled and installed in the target root and not `BROOT`, so it wouldn't be usable for building the package.

I think the easiest solution is to simply drop `MULTILIB_USEDEP` as was done with pkgconfig and just expect the user built Rust with all appropriate targets, but @leio wanted to the keep the multilib deps (at least at the time a couple years ago), so keeping EAPI 6 was the compromise to allow librsvg to both cross-compile and depend on multilib Rust.  Maybe you could check with @leio to see if he feels differently now.

I also tried to remove multilib stuff from Rust in #20727 which would get around this, but I don't see that being applied.

If `MULTILIB_USEDEP` can't be dropped, it's probably "most correct" to leave it in `BDEPEND` and hitting [bug #723112](https://bugs.gentoo.org/723112).  I'm fine with that being applied, I will fix it locally.

**Comment by SoapGentoo** [comment]
@dm0- could you update your branch so we could start doing rust like we do toolchain? I absolutely think your approach is the correct way forward

**Pull request closed without merging**